packages/react/src/portal/Portal.tsx

```diff
@@ -0,0 +1,66 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { FloatingPortal } from '@floating-ui/react';
+import { usePortalContext } from './PortalContext';
+import { HTMLElementType, refType } from '../utils/proptypes';
+
+/**
+ * @ignore - internal component.
+ */
+const Portal: Portal = function Portal(props: Portal.Props) {
+  const { children, container, keepMounted = false } = props;
+
+  const mounted = usePortalContext();
+
+  const shouldRender = mounted || keepMounted;
+  if (!shouldRender) {
+    return null;
+  }
+
+  return <FloatingPortal root={container}>{children}</FloatingPortal>;
+};
+
+namespace Portal {
+  export interface Props {
+    children?: React.ReactNode;
+    /**
+     * The container to render the portal element into.
+     */
+    container?: HTMLElement | null | React.RefObject<HTMLElement | null>;
+    /**
+     * Whether to keep the portal mounted in the DOM when the popup is closed.
+     * @default false
+     */
+    keepMounted?: boolean;
+  }
+
+  export interface State {}
+}
+
+interface Portal {
```

Comment:
Why is this needed?

Reply:
Prevent Root components from accepting invalid values